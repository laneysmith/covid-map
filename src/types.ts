export const COLORS = [
    '#DEEDCF',
    '#B8DEAA',
    '#88CE86',
    '#62BE73',
    '#3FAC6C',
    '#1D9A6C',
    '#188977',
    '#137177',
    '#0E4D64',
    '#0A2F51',
];

export enum Variable {
    CASES = 'cases',
    DEATHS = 'deaths',
}

export type ColorScales = { [key in Variable]: [] };

export type FipsStats = { [key in Variable]: number }

export type ResponseData = { data: object, maxCases: number, maxDeaths: number }
