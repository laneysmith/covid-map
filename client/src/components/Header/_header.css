header {
  position: absolute;
  left: 10px;
  top:10px;
  z-index: 1;
  text-shadow: 1px 1px 3px #FFFFFF, 1px 1px 1px #FFFFFF;
}

header h1 {
  font-size: 1.8rem;
  margin: 0 0 0.5rem 0;
}

header h2 {
  font-size: 1.4rem;
  margin: 0.5rem 0;
}
