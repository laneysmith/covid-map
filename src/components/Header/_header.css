header {
  position: absolute;
  left: 0px;
  top: 0px;
  width: 100%;
  z-index: 1;
}

header h1 {
  font-size: 1.8rem;
  margin: 0 0 0.5rem 0.5rem;
  text-shadow: 1px 1px 3px #ffffff, 1px 1px 1px #ffffff;
}

header h2 {
  font-size: 1.4rem;
  margin: 0 0 0.5rem 0.5rem;
  text-shadow: 1px 1px 3px #ffffff, 1px 1px 1px #ffffff;
}

.error-message {
  background: rgb(222,75,75);
  background: linear-gradient(135deg, rgba(222,75,75,1) 18%, rgba(255,114,114,1) 100%);
  color: white;
  font-size: 1rem;
  margin-bottom: 1rem;
  padding: 0.8rem 0.5rem;
  width: 100%;
  transition: height 1s ease-out; 
}

.error-message--hide {
  height: 0;
}

.error-message--show {
  height: unset;
}
