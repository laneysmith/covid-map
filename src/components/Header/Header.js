import React from 'react';

import './_header.css';

const Header = ({ selectedVariable, selectedDate }) => (
  <header>
    <h1>US Covid-19 {selectedVariable}</h1>
    <h2>{selectedDate}</h2>
  </header>
);

export default Header;
