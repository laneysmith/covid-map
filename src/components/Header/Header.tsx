import React from 'react';

import { Variable } from '../../types';
import './_header.css';

interface IHeader {
  selectedVariable: Variable;
  selectedDate?: string;
  error: boolean;
}

const Header: React.SFC<IHeader> = ({ selectedVariable, selectedDate, error }) => (
  <header>
    {error && <div className="error-message">Something went wrong. ¯\_(ツ)_/¯</div>}
    <h1>US Covid-19 {selectedVariable}</h1>
    {selectedDate && <h2>{selectedDate}</h2>}
  </header>
);

export default Header;
