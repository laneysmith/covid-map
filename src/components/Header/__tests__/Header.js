import React from 'react';
import { render } from '@testing-library/react';

import Header from '../Header';
import { Variable } from '../../../types';

describe('Header', () => {
  it('renders correctly without error banner', () => {
    const { container } = render(
      <Header selectedVariable={Variable.DEATHS} selectedDate="2020-02-14" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with error banner', () => {
    const { container } = render(
      <Header selectedVariable={Variable.DEATHS} selectedDate={null} error />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
