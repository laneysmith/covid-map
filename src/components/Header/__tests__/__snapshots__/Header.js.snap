// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly with error banner 1`] = `
<header>
  <div
    class="error-message"
  >
    Something went wrong. ¯\\_(ツ)_/¯
  </div>
  <h1>
    US Covid-19 
    deaths
  </h1>
</header>
`;

exports[`Header renders correctly without error banner 1`] = `
<header>
  <h1>
    US Covid-19 
    deaths
  </h1>
  <h2>
    2020-02-14
  </h2>
</header>
`;
